tests/moduletool/test_install.py

```diff
@@ -941,3 +942,386 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
+            """,
+            autostd=False,
+            python_package_sources=[local_module_package_index, index.url],
+            python_requires=[Requirement.parse(module.ModuleV2Source.get_package_name_for("parent_module"))],
+            install_project=True,
+        )
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "No matching distribution found for inmanta-module-child-module==3.3.3",
+    )
+
+    shutil.rmtree(os.path.join(str(tmpdir), "parent_module"))
+
+    # Scenario 2
+    # parent_module requires child_module v3.3.3 which is installed with v1.1.1
+
+    # Prepare the required module with a low version:
+
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "child_module"),
+        new_version=version.Version("1.1.1"),
+        new_name="child_module",
+        install=False,
+        publish_index=index,
+    )
+    parent_module = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "parent_module"),
+        new_version=version.Version("1.2.3"),
+        new_name="parent_module",
+        install=False,
+        new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
+        publish_index=index,
+    )
+
+    with pytest.raises(ModuleNotFoundException):
+        snippetcompiler_clean.setup_for_snippet(
+            f"""
+            import {module.ModuleV2.get_name_from_metadata(parent_module)}
+            """,
+            autostd=False,
+            python_package_sources=[local_module_package_index, index.url],
+            python_requires=[Requirement.parse(module.ModuleV2Source.get_package_name_for("parent_module"))],
+            install_project=True,
+        )
+
+    log_contains(
+        caplog,
+        "inmanta.env",
+        logging.DEBUG,
+        "No matching distribution found for inmanta-module-child-module==3.3.3",
+    )
+
+    shutil.rmtree(os.path.join(str(tmpdir), "parent_module"))
+    shutil.rmtree(os.path.join(str(tmpdir), "child_module"))
+
+    # Scenario 3
+    # parent_module requires child_module v3.3.3 which is installed with v3.3.3
+
+    # Prepare the required module with the correct version:
+    module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "child_module"),
+        new_version=version.Version("3.3.3"),
+        new_name="child_module",
+        install=False,
+        publish_index=index,
+    )
+
+    parent_module = module_from_template(
+        os.path.join(modules_v2_dir, "minimalv2module"),
+        os.path.join(str(tmpdir), "parent_module"),
+        new_version=version.Version("1.2.3"),
+        new_name="parent_module",
+        install=False,
+        new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
+        publish_index=index,
+    )
```

Comment:
This one was already published to the index earlier. No need to push it again.
```suggestion
    parent_module = module_from_template(
        os.path.join(modules_v2_dir, "minimalv2module"),
        os.path.join(str(tmpdir), "parent_module"),
        new_version=version.Version("1.2.3"),
        new_name="parent_module",
        install=False,
        new_requirements=[InmantaModuleRequirement.parse("child_module==3.3.3")],
        publish_index=index,
    )
```